### PR TITLE
[Refactor] Replace `torch.save` and `torch.load` with `InMemoryDataset.save` and `InMemoryDataset.load`

### PR DESCRIPTION
**Pull request by EdisonLeeeee**
A follow-up PR of  #7250

### CODE REVIEW

File: torch_geometric/datasets/actor.py [hunk 1]
```diff
@@ -50,7 +50,7 @@ class Actor(InMemoryDataset):
     def __init__(self, root: str, transform: Optional[Callable] = None,
                  pre_transform: Optional[Callable] = None):
         super().__init__(root, transform, pre_transform)
-        self.data, self.slices = torch.load(self.processed_paths[0])
+        self.load(self.processed_paths[0], data_cls=Data)
```
Comment (akihironitta) [inline torch_geometric/datasets/actor.py:53]: nit: do we need to specify `data_cls`?
```suggestion
        self.load(self.processed_paths[0])
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -20,6 +20,7 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/).
 
 - Disallow the usage of `add_self_loops=True` in `GCNConv(normalize=False)` ([#8210](https://github.com/pyg-team/pytorch_geometric/pull/8210))
 - Disable device asserts during `torch_geometric.compile` ([#8220](https://github.com/pyg-team/pytorch_geometric/pull/8220))
+- Replace `torch.save` and `torch.load` with `InMemoryDataset.save` and `InMemoryDataset.load` ([#8317](https://github.com/pyg-team/pytorch_geometric/pull/8317))
```
Comment (akihironitta) [inline CHANGELOG.md:23]: (nit: IMHO we don't need a changelog entry for internal refactor that doesn't impact users, but feel free to ignore my comment :))

```suggestion
```

**Review by akihironitta — APPROVED** [review]
Neat! ✨

File: torch_geometric/datasets/actor.py (hunk 1 above)
Comment (EdisonLeeeee) [reply torch_geometric/datasets/actor.py:53]: Good catch! Let's drop it.

File: torch_geometric/datasets/rel_link_pred_dataset.py [hunk 3]
```diff
@@ -52,7 +52,7 @@ def __init__(self, root: str, name: str,
         self.name = name
         assert name in ['FB15k-237']
         super().__init__(root, transform, pre_transform)
-        self.data, self.slices = torch.load(self.processed_paths[0])
+        self.load(self.processed_paths[0])
```
Comment (rusty1s) [inline torch_geometric/datasets/rel_link_pred_dataset.py:55]: here again.

**Comment by EdisonLeeeee** [comment]
@rusty1s Thank you for the fix!

**Pull request merged**